src/ms-graph/client.ts

```diff
@@ -149,15 +149,26 @@ export class GraphClient {
       delay: 30_000,
       timeout: 360_000,
       factor: 2,
-      handleError: (error, attemptContext) => {
+      handleError: async (error, attemptContext) => {
         if ([404, 403, 401].includes(error.statusCode)) {
           attemptContext.abort();
         }
 
         this.logger.info(
-          { error, attemptContext, link },
+          { error: JSON.stringify(error), attemptContext, link },
           `Encountered an error.`,
         );
+
+        if (error.statusCode === 429) {
+          const retryAfterSeconds = error.response?.headers?.['Retry-After'];
```

Comment:
From what I can tell, this value was 9 minutes in the last error. Although this is a while, it'll get us another 1500 requests to use.

Reply:
Do we want a default value for any instances where they fail to provide us with a Retry-After value?

Reply:
I think at this point no. This is already within the retry logic so we'll get 3 attempts with a good delay before we will fail.